R/create.R

```diff
@@ -241,7 +241,10 @@ create_initial_conditions <- function(data, delays, rt_prior, generation_time, m
 #' from the posterior whilst the approximate method returns approximate samples. The approximate method is likely to return results 
 #' several order of magnitudes faster than the exact method.
 #' @param verbose Logical, defaults to `FALSE`. Should verbose progress messages be returned.
-#'
+#' @param backend A character string defaults to "rstan". The backend to use for
+#'     computation.
+#' @param cache_model Logical, defaults to `TRUE`. Should a local copy be made 
+#'     and compiled of the stan program.
```

Comment:
I'd suggest moving these two new arguments (`backend` and `cache_model`) to the user-specified `stan_args` list, which is already being filtered during argument processing for each algorithm:
```
stn_args = list(
  backend = "cmdstan",
  cache_model = TRUE
)
```

Reply:
This might be a neat option and is perhaps what I should have done for the NUTs vs VI switch. What do you think @medewitt.? Does it introduce to much internal hassle/ would it be too hard to document?

Reply:
Good point! I wanted to comment on the `method` argument too. It could be moved to `stan_args` too and rename the methods as well since `exact` doesn't make sense with Bayesian :)
Here's what would be the default:
```
stn_args = list(
  backend = "rstan",
  method = "NUTS",
  ...
)
```
So, the method defaults to `method = "NUTS"` but can also accept `"method = "VB""`.

I can make the change if submit a PR, if you wish.

Reply:
Im a bit undecided. My concern is it places a lot into this list which may be hard to document/flag to the user. For less technical users they may totally miss these features.

Reply:
I see. You may document it in the `INFO` logs and in vignettes. For example, you could print [here](https://github.com/epiforecasts/EpiNow2/blob/d71ae94032927523d4becd014443a5f228aa27f9/R/estimate_infections.R#L406-L409) that `NUTS` (default) method will be used and that any of the supported methods (`method = c("NUTS", "VB")`) can be selected in the `stan_args` list.

Reply:
Yes good point. Thoughts @medewitt?

Reply:
@seabbs @hsbadr I definitely like the move to the `stan_args` list. Will need to tweak the `create_*` functions to set the optional args to NULL before passing them to `do.call` otherwise rstan/cmdstanr will yell at you. But this abstracts away these options and keeps the initial functions relatively clean.

I agree on mentioning it in the vignettes. I could see one vignette devoted to the different backends and estimation methods.